tests/conftest.py

```diff
@@ -6,10 +6,51 @@
 
 import pytest
 import dbusmock
+import os
+import sys
+import tempfile
 
 from tests import PortalMock
 
 
+def pytest_configure():
+    ensure_umockdev_loaded()
+    create_test_dirs()
+
+
+def ensure_umockdev_loaded():
+    umockdev_preload = "libumockdev-preload.so"
+    preload = os.environ.get("LD_PRELOAD", "")
+    if umockdev_preload not in preload:
+        os.environ["LD_PRELOAD"] = f"{umockdev_preload}:{preload}"
+        os.execv(sys.executable, [sys.executable] + sys.argv)
+
+
+def create_test_dirs():
+    test_root = tempfile.TemporaryDirectory(
+        prefix='xdp-testroot-',
+        ignore_cleanup_errors=True
+    )
+
+    env_dirs = [
+        'HOME',
+        'TMPDIR',
+        'XDG_CACHE_HOME',
+        'XDG_CONFIG_HOME',
+        'XDG_DATA_HOME',
+        'XDG_RUNTIME_DIR',
+    ]
+    for env_dir in env_dirs:
+        directory = os.path.join(test_root.name, env_dir.lower())
+        os.mkdir(directory, mode=0o700)
+        os.environ[env_dir] = directory
+        print('Setup', env_dir, 'as', directory)
```

Comment:
```suggestion
        directory = test_root / env_dir.lower()
        directory.mkdir(mode=0o700, parents=True)
        os.environ[env_dir] = directory.absolute()
        print('Setup', env_dir, 'as', directory)
```
because pathlib is nicer and iirc `tempfile` gives a Path object anyway.4


Does the `print` actually do something? iirc printing during conftest was a bit tricky (and arguably should be logger anyway)

Reply:
Removed the printing as well. It adds no value.